src/core/mac/mac_frame.cpp

```diff
@@ -116,6 +116,12 @@ void TxFrame::Info::PrepareHeadersIn(TxFrame &aTxFrame) const
         // | 13 | Extended     | Short        | Present      | Not Present  |      1       |
         // | 14 | Short        | Short        | Present      | Not Present  |      1       |
         // +----+--------------+--------------+--------------+--------------+--------------+
+        //
+        // This table shows the combination of flags allowed in an encoded MAC
+        // header. Regarding rows 9-14, when both Source and Destination
+        // Address fields are present and at least one uses a short address
+        // format, then if the source and destination PAN IDs are equal, PAN
+        // ID compression is set to 1.
```

Comment:
It looks like this conflicts with the text in the 802.15.4 standard(802154-2015 & 802154-2020), where it states:
> If both the destination and source addressing information is present and either is a short address, the MAC sublayer
shall compare the destination and source PAN IDs and the PAN ID Compression field shall be set to zero **if and
only if** the PAN identifiers are identical.

The precondition is `both the destination and source addressing information is present and either is a short address`. Then the compression bit should be set to `0` **if and only if** source PAN ID and destination PAN ID  are identical. Here's my interpretation.

| Source PAN ID | Destination PAN ID | Condition |PAN ID Compression |
| ------------- | ------------------ | -- | ------------------- | 
|Present | Present | Source PAN ID == Destination PAN ID| 0 |
|Present|Present | Source PAN ID != Destination PAN ID| 1 |
|Present|Not Present|  | 1 |
|Not Present|Present| | 1 |

@jwhui @zhanglongxia

Reply:
When we parse a frame, if the PAN ID compression bit is set to 1, how do we know if the source PAN ID is present in your case?

Reply:
Yes. that's the problem. So there may be an issue in the 802.15.4 standard.

Reply:
There were corrections/clarifications regarding this:
- [Proper PAN ID Field Settings for 802.15.4-2015>](https://mentor.ieee.org/802.15/dcn/15/15-15-0911-01-0mag-proper-pan-id-field-settings-for-802-15-4-2015.docx)
- [Yet another document explaining PAN ID Compression](https://mentor.ieee.org/802.15/dcn/15/15-15-0561-03-0mag-yet-another-document-explaining-pan-id-compression.docx)

The second doc is mentions:
> When the Source Address field is present and the Destination address field is present and one or both of source address or destination address fields contain a short address
> - IF the source and destination PAN IDs are equal, PAN ID compression is set to 1 and the PAN ID field contains the (one) PAN ID (according to figure 35 of 2011, the Source PAN Identifier field is present and the Destination PAN Identifier field is absent)
> - IF the source and destination IDs are NOT EQUAL, the PAN ID compression field is set to ZERO and BOTH source and destination PAN ID fields are present in the frame.

Reply:
Thanks. It's interesting. I checked 802.15.4-2020, and it doesn't include these corrections.

Reply:
IEEE 802.15.4-2020 states the following:

```
If both destination and source addressing information is present, the MAC sublayer shall compare
the destination and source PAN identifiers. If the PAN IDs are identical, the PAN ID Compression
field shall be set to one, and the Source PAN ID field shall be omitted from the transmitted frame. If
the PAN IDs are different, the PAN ID Compression field shall be set to zero, and both Destination
PAN ID field and Source PAN ID fields shall be included in the transmitted frame.
```

Reply:
> IEEE 802.15.4-2020 states the following:
> 
> ```
> If both destination and source addressing information is present, the MAC sublayer shall compare
> the destination and source PAN identifiers. If the PAN IDs are identical, the PAN ID Compression
> field shall be set to one, and the Source PAN ID field shall be omitted from the transmitted frame. If
> the PAN IDs are different, the PAN ID Compression field shall be set to zero, and both Destination
> PAN ID field and Source PAN ID fields shall be included in the transmitted frame.
> ```

I think this block is for version `0b00` and `0b01`. The table in question is for version `0b10`.

Reply:
For Version 2 frames, the table is not consistent with the footnote. The last row of the table that sets PAN ID Compression to 1 would not make any sense if the Destination PAN ID is not equal to the Source PAN ID.